resources/assets/components/CampaignInbox/index.js

```diff
@@ -65,6 +67,40 @@ class CampaignInbox extends React.Component {
     });
   }
 
+  updateQuantity(post, newQuantity) {
+    // Fields to send to /posts
+    const fields = {
+      northstar_id: post.user.id,
+      campaign_id: post.signup.campaign_id,
+      campaign_run_id: post.signup.campaign_run_id,
+      quantity: newQuantity,
+    };
+
+    // Make API request to Rogue to update the quantity on the backend
+    let response = this.api.post('api/v2/posts', fields);
+
+    response.then((result) => {
+      // Update the state
+      this.setState((previousState) => {
+        const newState = {...previousState};
+        const signupChanged = post.signup_id;
+
+        // Update the quantity for each post under this signup
+        forEach (newState.posts, (value) => {
```

Comment:
@sbsmith86 we end up mapping things out to be all of the posts that hold the data for their signup since we need to be able to map through the posts to pass them to the `InboxItem` component. The way this works is pretty confusing to me so I'm definitely down to discuss it